client/app/components/compose.coffee

```diff
@@ -52,6 +52,10 @@ module.exports = Compose = React.createClass
 
 
     shouldComponentUpdate: (nextProps, nextState) ->
+        if @state.saving isnt nextState.saving
```

Comment:
The problem is : 
 * Application start automatic saving
 * You add a contact  : @state is updated
 * Application finish automatic saving
 * New contact isn't in nextState , thus contact just added disappears.